File: common/node/api/src/Device.ts
import AdditionalState from "./AdditionalState";
import BaseDevice from "./BaseDevice";
import Battery from "./Battery";
import Capability from "./Capability";
import DeviceState from "./DeviceState";

export default interface Device extends BaseDevice, Battery {
    state: DeviceState;
    since: number;
    additionalState?: AdditionalState;
    capability?: Capability;
}

File: common/go/models/device.go
package models

type BaseDevice struct {
	Name        string `json:"name"`
	DisplayName string `json:"display_name"`
	Type        string `json:"type"`
	Location    string `json:"location"`
	Visible     bool   `json:"visible"`
}

type Device struct {
	BaseDevice
}

type DeviceState string

const (
	On  DeviceState = "on"
	Off DeviceState = "off"
)

type AdditionalState struct {
	Brightness *int `json:"brightness,omitempty"`
}

type Capability struct {
	Brightness bool `json:"brightness,omitempty"`
}
